Mathlib/Combinatorics/DyckWord.lean

```diff
@@ -0,0 +1,424 @@
+/-
+Copyright (c) 2024 Jeremy Tan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jeremy Tan
+-/
+import Mathlib.Combinatorics.Catalan
+import Mathlib.Data.List.Indexes
+
+/-!
+# Dyck words
+
+## Main definitions
+
+* `DyckPath`: a list of `Step`s, each of which may only be `U`p or `D`own.
+* `IsBalanced`: the property defining Dyck words. Translating `U` and `D` to `(` and `)`,
+  the resulting brackets are balanced.
+
+## Main results
+
+* `treeEquiv`: equivalence between Dyck words and rooted binary trees.
+* `finiteTreeEquiv`: equivalence between Dyck words of length `2 * n` and rooted binary trees
+  with `n` internal nodes.
+* `treesOfNodesEq_card_eq_catalan`: there are `catalan n` Dyck words of length `2 * n`.
+-/
+
+
+/-- A `Step` is either `U` or `D`, corresponding to `(` and `)` respectively. -/
+inductive Step
```

Comment:
This should be in a namespace, currently this creates a type in the global namespace called `Step` which is way too generic a a name.